showkase-processor/src/main/java/com/airbnb/android/showkase/processor/ShowkaseProcessor.kt

```diff
@@ -117,19 +117,18 @@ class ShowkaseProcessor: AbstractProcessor() {
         }.toSet()
 
     private fun processPreviewAnnotation(roundEnvironment: RoundEnvironment): Set<ShowkaseMetadata> {
-        val previewClass = Class.forName(PREVIEW_CLASS_NAME)
-        val previewClassAnnotation = (previewClass as Class<out Annotation>)
-        val previewTypeMirror = elementUtils
-            .getTypeElement(previewClass.canonicalName)
-            .asType()
-        return roundEnvironment.getElementsAnnotatedWith(previewClassAnnotation).mapNotNull { element ->
-            showkaseValidator.validateComponentElement(element, composableTypeMirror, typeUtils, 
-                previewClass.simpleName)
-            val showkaseMetadata = getShowkaseMetadataFromPreview(
-                element as ExecutableElement, elementUtils, typeUtils, previewTypeMirror, showkaseValidator
-            )
-            showkaseMetadata
-        }.toSet()
+        val previewTypeElement = elementUtils.getTypeElement(PREVIEW_CLASS_NAME)
```

Comment:
This was updated because using `Class.forName` directly was causing a `ClassNotFoundException`. After asking around, I was advised to directly use `elementUtils.getTypeElement` instead of using the canonical name of `Class.forName` which effectively does the same thing. 

However, I needed to add a null check to this previewTypeElement. All this is working fine even with the null check but I'm wondering in which situations does it return null and why? If I remove the null check, I get an NPE.

Reply:
From what I can tell compose or the ui utils is not a dependency of this processor (not in build.gradle), which means you can't access it as a class. It will be dependent on whether the thing you are processing has that class present, which is why it is correct to access it in a nullable way, as you have to support any case of your processor being run in whatever environment, and can't make assumption about what will be present (or you have to add the dependency to the processor's build.gradle)

Reply:
:O For some reason I remembered having all compose dependencies for showkase_testing. I might've cleaned it up at some point :O Should've just double checked. I think this is in fact doing the right thing 🥇